src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -430,16 +430,29 @@ async def infer_runtime_platforms(request: RuntimePlatformsRequest) -> RuntimePl
         file_name = next(
             rt.file_name()
             for rt in request.runtime.known_runtimes
-            if version == (rt.major, rt.minor)
-            and (request.architecture is None or request.architecture.value == rt.architecture)
+            if version == (rt.major, rt.minor) and request.architecture.value == rt.architecture
         )
     except StopIteration:
-        # Not a known runtime, so fallback to just passing a platform
-        # TODO: maybe this should be an error, and we require users to specify
-        # complete_platforms themselves?
-        return RuntimePlatforms(
-            interpreter_version=version,
-            pex_platforms=PexPlatforms((_format_platform_from_major_minor(*version),)),
+        # Not a known runtime, so error out.
+        raise ValueError(
+            softwrap(
+                f"""\
+                Could not find a known runtime for the given Python version and machine architecture!
+
+                Python version: {version}
```

Comment:
For the follow up, potentially this should indicate when the version is inferred from the ICs, e.g. 

```suggestion
                Could not find a known runtime for the [given/inferred] Python version and machine architecture!

                Python version: {version} [(inferred from interpreter constraints)]
```

where the `[]` parts are optional based on whether the `_infer_from_ics` code-path is used or not.